### ISSUE TITLE
Problem uploading files in general

### ISSUE THREAD

**Issue by jayminer81** [punktniklas/NiKom#15]
Now when I try and upload a file, the file gets sent and everything works as it should until after I have typed in the description and pressed enter, then I get a: "Rexx: Return-code 10"-error message, and IconX (which I use to launch NiKom) pops up it's window and says "+++ Error 15 in line 5: Function not found".

The file gets into NiKom:Tempfiler/ but never gets moved to it's correct place, it get's listed in the BBS so the BBS "believes" the file is there but it can not be downloaded. I read somewhere in the documentation that a 'move'-command was being used and thought that might be it but I tried to download a 'move'-command and it made no difference.

**Comment by punktniklas** [comment]
It's true that moving of files is handled by an external command. It's available in the NiKom 1.60 archive in NiKomInstall/Move and needs to be placed somewhere in the path. The reason for this is that it handles recognizing when the from and to directory is on the same partition. In that case files can be moved with a simple rename operation and all the bytes don't have to be copied. If the destination is on a different partition it will do a full copy and delete. I was apparently to lazy to implement this internally in NiKom.

Now comes the tricky part. :-) Move is not launched directly by NiKom. Instead an ARexx script is launched which executes the move command. This ARexx script is Rexx/ExtKom10.nik.

I looked at this script in the 1.60 distribution and it seems like Tomas Kärki has put his own "hacked" version in the distribution which includes a lot of stuff for his "SRatio" package. I have no idea what all of this does but from your error message it sounds like it can't find the function "Word". (Don't know/remember what this is..)

The original ExtKom10.nik was a lot smaller. I've pasted it below. Try using the original and see if it works better. It should probably be replaced in the distribution files as well.

-----------------
/* Program för att flytta filer rätt */

parse arg nod pers from to .

address command 'move >NIL:' from to
